src/main/java/org/junit/rules/Timeout.java

```diff
@@ -163,6 +162,19 @@ public Statement apply(Statement base, Description description) {
     }
 
     /**
+     * Creates a {@link Statement} that will run the given
+     * {@code statement} in a thread with a name based on
+     * the given {@code description}, and timeout the operation
+     * based on the values configured in this rule.
+     */
+    private Statement createFailOnTimeoutStatement(
+            Statement statement, Description description) throws Exception {
+        return FailOnTimeout.builder()
+            .withDescription(description)
+            .build(createFailOnTimeoutStatement(statement));
```

Comment:
I'm confused... isn't this wrapping a `Statement` that applies the timeout inside of another `Statement` that applies a timeout?